Python/cpuinfo.c

```diff
@@ -0,0 +1,556 @@
+#include "pycore_cpuinfo.h"
+
+/* CPUID input and output registers are 32-bit unsigned integers */
+#define CPUID_REG                   uint32_t
+/* Check one or more CPUID register bits. */
+#define CHECK_REG(REG, MASK)        ((((REG) & (MASK)) == (MASK)) ? 0 : 1)
+#define CPUID_CHECK_REG(REG, FEAT)  CHECK_REG(REG, (Py_CPUID_MASK_ ## FEAT))
+#define XSAVE_CHECK_REG(REG, FEAT)  CHECK_REG(REG, (Py_XSAVE_MASK_ ## FEAT))
+
+// For now, we only try to enable SIMD instructions for x86-64 Intel CPUs.
+// In the future, we should carefully enable support for ARM NEON and POWER
+// as well as AMD.
+#if defined(__x86_64__) && defined(__GNUC__)
+#  include <cpuid.h>      // __cpuid_count()
+#  define HAS_CPUID_SUPPORT
+#  define HAS_XGETBV_SUPPORT
+#elif defined(_M_X64)
+#  include <immintrin.h>  // _xgetbv()
+#  define HAS_XGETBV_SUPPORT
+#  include <intrin.h>     // __cpuidex()
+#  define HAS_CPUID_SUPPORT
+#else
+#  undef HAS_CPUID_SUPPORT
+#  undef HAS_XGETBV_SUPPORT
+#endif
+
+// Below, we declare macros for guarding the detection of SSE, AVX/AVX2
+// and AVX-512 instructions. If the compiler does not even recognize the
+// corresponding flags or if we are not on an 64-bit platform we do not
+// even try to inspect the output of CPUID for those specific features.
+#ifdef HAS_CPUID_SUPPORT
+#if defined(Py_CAN_COMPILE_SIMD_SSE_INSTRUCTIONS)              \
+    || defined(Py_CAN_COMPILE_SIMD_SSE2_INSTRUCTIONS)          \
+    || defined(Py_CAN_COMPILE_SIMD_SSE3_INSTRUCTIONS)          \
+    || defined(Py_CAN_COMPILE_SIMD_SSSE3_INSTRUCTIONS)         \
+    || defined(Py_CAN_COMPILE_SIMD_SSE4_1_INSTRUCTIONS)        \
+    || defined(Py_CAN_COMPILE_SIMD_SSE4_2_INSTRUCTIONS)        \
+    // macros above should be sorted in alphabetical order
+#  define SIMD_SSE_INSTRUCTIONS_DETECTION_GUARD
+#endif
+
+#if defined(Py_CAN_COMPILE_SIMD_AVX_INSTRUCTIONS)               \
+    || defined(Py_CAN_COMPILE_SIMD_AVX_IFMA_INSTRUCTIONS)       \
+    || defined(Py_CAN_COMPILE_SIMD_AVX_NE_CONVERT_INSTRUCTIONS) \
+    || defined(Py_CAN_COMPILE_SIMD_AVX_VNNI_INSTRUCTIONS)       \
+    || defined(Py_CAN_COMPILE_SIMD_AVX_VNNI_INT8_INSTRUCTIONS)  \
+    || defined(Py_CAN_COMPILE_SIMD_AVX_VNNI_INT16_INSTRUCTIONS) \
+    // macros above should be sorted in alphabetical order
+#  define SIMD_AVX_INSTRUCTIONS_DETECTION_GUARD
+#endif
+
+#if defined(Py_CAN_COMPILE_SIMD_AVX2_INSTRUCTIONS)              \
+    // macros above should be sorted in alphabetical order
+#  define SIMD_AVX2_INSTRUCTIONS_DETECTION_GUARD
```

Comment:
```suggestion
#if defined(Py_CAN_COMPILE_SIMD_AVX2_INSTRUCTIONS)
    // macros above should be sorted in alphabetical order
#  define SIMD_AVX2_INSTRUCTIONS_DETECTION_GUARD
```
The `#define` statement is already on a newline, so it does not need to be continued.